python/tvm/driver/build_module.py

```diff
@@ -277,7 +277,7 @@ def _build_for_device(input_mod, target, target_host):
                 lambda f: "calling_conv" not in f.attrs
                 or f.attrs["calling_conv"].value != CallingConv.DEVICE_KERNEL_LAUNCH
             ),
-            tvm.tir.transform.Apply(lambda f: f.with_attr("target", target)),
+            tvm.tir.transform.Apply(lambda f: f.with_attr("target", target_host)),
```

Comment:
For the record, segfault with nvptx was happening because the generated host code was calling intrinsics registered for nvptx, like `__nv_log2` or `__nv_ceil`. The reason it was working on CUDA was just by coincident: there is no CUDA intrinsics registered for fp64 log2, ceil.

This change fixes the issue I mentioned above.